### PR TITLE
Feature: Add more messages to chat hider

### PR DESCRIPTION
**Pull request by martimavocado**
adds the following messages to the annoying category:
- not enough breaking power for gems
- useless messages for skymall and jacob's event artifact

adds a toggle to hide skymall perk message outside mining islands
adds a toggle to hide jacob's event artifact message outside garden

adds .inMiningIsland to LocationUtils

### CODE REVIEW

**Comment by lunaynx** [comment]
Nitpick: I'm pretty sure Sky Mall won't work in Gold Mine or Deep Caverns, so should be hidden there too. (You would probably only be there for bestiary, or exp share cores if Ironman, after unlocking the Dwarven Mines).

**Comment by martimavocado** [comment]
> I'm pretty sure Sky Mall won't work in Gold Mine or Deep Caverns, so should be hidden there too.

Some perks still work outside Dwarven Mines, but the ones that don't, also don't work in crystal hollows from the testing i did a few weeks ago

File: src/main/java/at/hannibal2/skyhanni/utils/LocationUtils.kt [hunk 1]
```diff
@@ -39,4 +41,7 @@ object LocationUtils {
         val inFov = true // TODO add Frustum "Frustum().isBoundingBoxInFrustum(entity.entityBoundingBox)"
         return noBlocks && notTooFar && inFov
     }
+
+    fun inMiningIsland() = IslandType.GOLD_MINES.isInIsland() || IslandType.DEEP_CAVERNS.isInIsland()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/utils/LocationUtils.kt]: This belongs in LorenzUtils. LocationUtils is currently only for utils functions around LorenzVec, not for skyblock related functions. We should maybe change this in the future and move all "isinSkyBlock" etc stuff around, but not in this pr.

File: src/main/java/at/hannibal2/skyhanni/features/chat/ChatFilter.kt [hunk 2]
```diff
@@ -253,6 +258,19 @@ class ChatFilter {
...
+    )
+
+    private val anitaFortunePattern by RepoPattern.pattern(
+        "chat.jacobevent.accessory",
+        "§e\\[NPC] Jacob§f: Your §9Anita's (\\w+) §fis giving you §6\\+(\\d{1,2})☘ (\\w+) Fortune §fduring the contest!"
+    )
+
+    private val skymallPerkPattern by RepoPattern.pattern(
+        "chat.skymall.perk",
+        "§eNew buff§r§r§r:(.*)"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/chat/ChatFilter.kt:273]: hypixel moment

**Review by hannibal002 — CHANGES_REQUESTED** [review]

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**